xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAEnumeratorHD.cpp

```diff
@@ -733,3 +545,227 @@ ComPtr<ID3D11VideoProcessorOutputView> CEnumeratorHD::CreateVideoProcessorOutput
+  if (isSourceFullRange)
+    return {};
+
+  // Top left chroma siting is always preferred for HDR because is used in UHD Blu-Rays
+  // ffmpeg flags of the source can be missing of bad anyway
+  DXGI_COLOR_SPACE_TYPE inputCS = DXGI_COLOR_SPACE_YCBCR_STUDIO_G22_TOPLEFT_P2020;
+
+  const DXGI_COLOR_SPACE_TYPE destColor = DX::Windowing()->UseLimitedColor()
+                                              ? DXGI_COLOR_SPACE_RGB_STUDIO_G22_NONE_P709
+                                              : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709;
+  const DXGI_COLOR_SPACE_TYPE outputCS = destColor;
+
+  const ProcessorConversions convTopLeft =
+      ListConversions(m_input_dxgi_format, std::vector<DXGI_COLOR_SPACE_TYPE>{inputCS},
+                      RenderingOutputFormats, std::vector<DXGI_COLOR_SPACE_TYPE>{outputCS});
+
+  CLog::LogF(LOGDEBUG, "BT.2020 input color spaces supported with SDR {} range output: {}: {}",
+             DX::Windowing()->UseLimitedColor() ? "limited" : "full",
+             DX::DXGIColorSpaceTypeToString(inputCS), !convTopLeft.empty() ? "yes" : "no");
+
+  if (!convTopLeft.empty())
+  {
+    if (CServiceBroker::GetLogging().IsLogLevelLogged(LOGDEBUG) &&
+        CServiceBroker::GetLogging().CanLogComponent(LOGVIDEO))
+    {
+      for (const auto& c : convTopLeft)
+        CLog::LogF(LOGDEBUG, "supported conversion: {}", c.ToString());
+    }
+
+    CLog::LogF(LOGDEBUG, "Preferred UHD Blu-Ray top left chroma siting will be used.");
+    return convTopLeft;
+  }
+
+  // Try left chroma siting as fall back - half a pixel is not much difference at 4k
+  inputCS = DXGI_COLOR_SPACE_YCBCR_STUDIO_G22_LEFT_P2020;
+  const ProcessorConversions convLeft =
+      ListConversions(m_input_dxgi_format, std::vector<DXGI_COLOR_SPACE_TYPE>{inputCS},
+                      RenderingOutputFormats, std::vector<DXGI_COLOR_SPACE_TYPE>{outputCS});
+
+  CLog::LogF(LOGDEBUG, "BT.2020 input color spaces supported with SDR {} range output: {}: {}",
+             DX::Windowing()->UseLimitedColor() ? "limited" : "full",
+             DX::DXGIColorSpaceTypeToString(inputCS), !convLeft.empty() ? "yes" : "no");
+
+  if (!convLeft.empty())
+  {
+    if (CServiceBroker::GetLogging().IsLogLevelLogged(LOGDEBUG) &&
+        CServiceBroker::GetLogging().CanLogComponent(LOGVIDEO))
+    {
+      for (const auto& c : convLeft)
+        CLog::LogF(LOGDEBUG, "supported conversion: {}", c.ToString());
+    }
+
+    CLog::LogF(LOGDEBUG, "Left chroma siting will be used.");
+    return convLeft;
+  }
+
+  return {};
+}
+
+ProcessorConversions CEnumeratorHD::QuerySDRConversions(const bool isSourceFullRange)
+{
+  std::unique_lock<CCriticalSection> lock(m_section);
+
+  // Not initialized yet
+  if (!m_pEnumerator)
+    return {};
+
+  const DXGI_COLOR_SPACE_TYPE inputCS = isSourceFullRange
+                                            ? DXGI_COLOR_SPACE_YCBCR_FULL_G22_LEFT_P709
+                                            : DXGI_COLOR_SPACE_YCBCR_STUDIO_G22_LEFT_P709;
```

Comment:
There at least one use case with old SD videos (DVD) not handled properly: these MPEG2 videos should use primaries P601 and color space `DXGI_COLOR_SPACE_YCBCR_STUDIO_G22_LEFT_P601`. As this is working properly in current master could be considered a regression, then please add here the code to handle these cases:

Mainly use colors space:

`DXGI_COLOR_SPACE_YCBCR_STUDIO_G22_LEFT_P601`
`DXGI_COLOR_SPACE_YCBCR_FULL_G22_LEFT_P601`

when primaries reported by FFmpeg are `AVCOL_PRI_BT470BG` or `AVCOL_PRI_SMPTE170M`  (see old code)

This is a sample video: 
[tsbuffer.zip](https://github.com/xbmc/xbmc/files/12066990/tsbuffer.zip)

Note that some times these old videos not has this information's properly signaled `AVCOL_PRI_UNSPECIFIED` then can be used video resolution to infer that is SD video. As this is not implemented in current code is not a blocking thing and can be implemented later. But RenderShaders has it implemented:

https://github.com/xbmc/xbmc/blob/022ef38abf96bdf8e2834259e0c147a9ef1161bb/xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererShaders.cpp#L195-L206

There also some special case with `DXGI_COLOR_SPACE_YCBCR_FULL_G22_NONE_P709_X601` (primaries BT.709 and tranfer BT.601) and RGB color spaces but again this can be done on a follow-up PR.

As a requirement I think it is enough to support regular SD videos here.

Reply:
> If the information is available, maybe extend with conditions on codec or container type, like mpeg-2, xvid, avi... all that

I vote for keep it simple same as is in PS for now. Codecs and containers not helps much... i.e. MPEG-2 is in both DVD and Blu-Ray and guess only when primaries are `AVCOL_PRI_UNSPECIFIED`

Reply:
Sure.

For P601/P709_X601, rather than duplicate the logic, I'm going to change the Query*Conversions to use AvToDxgiColorSpace(). I kept it around for that purpose for a future refactor, might as well do it now. It only needs minor changes. Will be a separate commit.

Reply:
Refactor of Query*Conversions to use AvToDxgiColorSpace() will come in another PR for clarity. Added the few missing lines to QuerySDRConversions() for now. Tested with your sample, with DVD (flags properly reported) > OK.

MS indicates that DXGI_COLOR_SPACE_YCBCR_FULL_G22_NONE_P709_X601 is for jpg, which Kodi won't decode with dxva, so the code will likely never be used.

Improvement for AVCOL_PRI_UNSPECIFIED requires touching other render methods to share the logic and code, so beyond the scope of this PR.